File: apps/server/src/routes/archetypes.ts
// Public, registry-derived archetype catalog. This is the API/SDK-codegen source
// for customer-selectable device + iOS + Safari combinations. Internal reference
// baselines and planned placeholders are deliberately excluded.

import type { FastifyInstance } from 'fastify';
import {
  ARCHETYPE_REGISTRY,
  LOCKED_ARCHETYPE_ID,
  type ListArchetypesResponse,
  type PublicArchetypeStatus,
} from '@driftstack/api-types';

export const ARCHETYPE_CATALOG_CACHE_MAX_AGE_SECONDS = 300;

const SELECTABLE_STATUSES = new Set<PublicArchetypeStatus>(['launch', 'available']);

const response: ListArchetypesResponse = {
  default_archetype_id: LOCKED_ARCHETYPE_ID,
  data: ARCHETYPE_REGISTRY.filter(
    (entry): entry is typeof entry & { status: PublicArchetypeStatus } =>
      SELECTABLE_STATUSES.has(entry.status as PublicArchetypeStatus),
  ).map((entry) => ({
    id: entry.id,
    display_label: entry.displayLabel,
    device: entry.device,
    ios_version: entry.iosVersion,
    safari_version: entry.safariVersion,
    canvas_family: entry.canvasFamily,
    status: entry.status,
    is_default: entry.id === LOCKED_ARCHETYPE_ID,
  })),
};

export function registerArchetypeRoutes(app: FastifyInstance): void {
  app.get('/v1/archetypes', (_request, reply) => {
    reply.header(
      'cache-control',
      `public, max-age=${ARCHETYPE_CATALOG_CACHE_MAX_AGE_SECONDS.toString()}`,
    );
    return response;
  });
}

File: packages/sdk-python/src/driftstack/resources/archetypes.py
"""Public browser archetype discovery — ``GET /v1/archetypes``."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

from driftstack.http import AsyncHttpClient, HttpClient, parse_model


class PublicArchetype(BaseModel):
    id: str
    display_label: str
    device: str
    ios_version: str
    safari_version: str
    canvas_family: Literal["A", "B"]
    status: Literal["launch", "available"]
    is_default: bool


class ListArchetypesResponse(BaseModel):
    default_archetype_id: str
    data: list[PublicArchetype]


class ArchetypesResource:
    """Synchronous server-authoritative archetype catalog."""

    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def list(self) -> ListArchetypesResponse:
        data = self._http.request("GET", "/v1/archetypes")
        return parse_model(ListArchetypesResponse, data)


class AsyncArchetypesResource:
    """Asynchronous server-authoritative archetype catalog."""

    def __init__(self, http: AsyncHttpClient) -> None:
        self._http = http

    async def list(self) -> ListArchetypesResponse:
        data = await self._http.request("GET", "/v1/archetypes")
        return parse_model(ListArchetypesResponse, data)
